lib/matchers/matchers.rb

```diff
@@ -247,26 +247,26 @@ def try_match(actual, op, expected) # rubocop:disable Metrics/CyclomaticComplexi
     # if actual and expected are strings
     if expected.is_a?(String) && actual.is_a?(String)
       return actual.casecmp(expected) == 0 if op == :==
-      return Gem::Version.new(actual).method(op).call(Gem::Version.new(expected)) if
```

Comment:
Could you comment as to why these changes are needed to get the tests running more cleanly?

Reply:
They're not. They're part of the general cleanup of stuff that I saw as I went through. The main reason this should go in is this:

```
2896 % ./cmp.rb 
Warming up --------------------------------------
         method+call   154.352k i/100ms
                send   248.925k i/100ms
Calculating -------------------------------------
         method+call      2.603M (± 1.4%) i/s -     13.120M in   5.041969s
                send      6.425M (± 3.1%) i/s -     32.111M in   5.003057s

Comparison:
                send:  6424944.9 i/s
         method+call:  2602626.5 i/s - 2.47x  slower
```

Especially considering that this is on the matcher side of things, and thus, hit by all users.